tor_ocr/main.py

```diff
@@ -41,29 +40,37 @@
     u_tor_post_id.reply(ocr_magic)
 """
 
+# "helloworld" is a valid API key, however use it sparingly
+__OCR_API_KEY__ = os.getenv('OCR_API_KEY', 'helloworld')
 
-def process_image(local_file):
-    with PyTessBaseAPI() as api:
-        api.SetImageFile(local_file)
-        text = api.GetUTF8Text()
 
-        confidences = api.AllWordConfidences()
-        if not confidences or len(confidences) == 0:
-            # we have an image, but it *really* couldn't find anything, not
-            # even false positives.
-            return None
+def process_image(image_url):
+    """
+    Processes an image with OCR, using ocr.space
+    :param image_url: a string url of what you need OCRed
+    :return: A dictionary containing several values, most importantly 'text'
+    """
+    json_result = decode_image_from_url(image_url)
 
-        logging.debug('Average of confidences: {}'.format(
-            sum(confidences) / len(confidences))
-        )
+    result = {
+        'text': json_result['ParsedText'],
```

Comment:
It will explode on this line if it is given an imgur album. Result from json_result:

```
(Pdb) json_result
{'ParsedResults': None, 'OCRExitCode': 3, 'IsErroredOnProcessing': True, 'ErrorMessage': ['File failed validation. Not a valid URL. URL must be of a valid and existing Image or PDF'], 'ErrorDetails': '', 'ProcessingTimeInMilliseconds': '78', 'SearchablePDFURL': None}
```
`image_url` == 'https://imgur.com/a/OdI7d'

Reply:
It looks like the response dict has changed from when this sample code was made. This is the response -- I... cleaned it a little because it's from r/4chan.

```
(Pdb) image_url
'https://i.redd.it/qhjbra3s92wz.png'
(Pdb) json_result
{'ParsedResults': [{'TextOverlay': {'Lines': [], 'HasOverlay': False, 'Message': 'Text overlay is not provided as it is not requested'}, 'FileParseExitCode': 1, 'ParsedText': 'Anonymous N0U0891922 [Reply] \r\nDaddy... \r\n', 'ErrorMessage': '', 'ErrorDetails': ''}], 'OCRExitCode': 1, 'IsErroredOnProcessing': False, 'ErrorMessage': None, 'ErrorDetails': None, 'ProcessingTimeInMilliseconds': '2568', 'SearchablePDFURL': ''}
```

Reply:
Yes, it will explode on Imgur albums, and then return `None`.

Reply:
Ah, you're right.

Reply:
Re: imgur albums: No, it completely explodes with a keyerror. Look at the JSON response I posted -- there is no `ParsedText` field.